src/store.js

```diff
@@ -468,8 +471,9 @@ export const appReducer = (state = initialState(), action) => {
 
     // SIDE EFFECTS: syncRemoteData, localStorage, updateUrlHistory
     existingItemChange: ({ oldValue, newValue, context, showContexts, itemsRanked, rankInContext, contextChain }) => {
-
-      if (oldValue === newValue) {
+      // second condition: exit so that when deleting a divider, it does not
+      // get merged into previous thought
+      if (oldValue === newValue || newValue === oldValue + '---') {
```

Comment:
Ideally this would be handled "higher up", i.e. prevent `existingItemChange` from getting dispatched at all. The check should happen where code is merged, probably in `deleteItem`.